scripts/scripts.js

```diff
@@ -56,6 +58,168 @@ function buildAutoBlocks(main) {
+  dialogContent.classList.add('modal-body');
+
+  const encodeLink = encodeURIComponent(link);
+  const encodeTitle = encodeURIComponent(document.title);
+
+  const socialPlatforms = [
+    {
+      name: 'Twitter',
+      icon: '/icons/twitter-icon.png',
+      shareUrl: `https://twitter.com/intent/tweet?url=${encodeLink}`,
+    },
+    {
+      name: 'Facebook',
+      icon: '/icons/facebook-icon.png',
+      shareUrl: `http://www.facebook.com/sharer.php?u=${encodeLink}&t=${encodeTitle},'sharer',toolbar=0,status=0,width=626,height=436`,
+    },
+    {
+      name: 'WhatsApp',
+      icon: '/icons/whatsapp-icon.png',
+      shareUrl: `https://api.whatsapp.com/send?text=Hey! Check out this: ${encodeLink}`,
+    },
+    {
+      name: 'LinkedIn',
+      icon: '/icons/linkedin-icon.png',
+      shareUrl: `https://www.linkedin.com/sharing/share-offsite/?url=${encodeLink}`,
+    },
+    { name: 'CopyLink', icon: '/icons/copy-link-icon.png' },
+  ];
+
+  socialPlatforms.forEach((platform) => {
+    const icon = document.createElement('img');
+    icon.src = platform.icon;
+    icon.alt = platform.name;
+    icon.classList.add('social-icon');
+    if (platform.shareUrl) {
+      icon.onclick = () => window.open(platform.shareUrl, '_blank');
+    } else if (platform.name === 'CopyLink') {
+      icon.onclick = () => {
+        navigator.clipboard.writeText(link);
+      };
+    }
+    dialogContent.appendChild(icon);
+  });
+  divContainer.appendChild(dialogContent);
+  dialog.appendChild(divContainer);
+  document.body.appendChild(dialog);
+
+  dialog.showModal();
+}
+
+/**
+ * Decorates the specified anchor elements with the 'social-share' class
+ * if they contain icon 'gray-share-icon.svg'.
+ * Adds a click event listener to each decorated anchor element.
+ * @param {Array<Element>} anchorElements - The anchor elements to decorate.
+ * @param {Element} block - The block element containing the modal.
+ * @param {Function} createSocialButton - The function to create the social button.
+ * @returns {void}
+ */
+function decorateSocialShare(anchorElements) {
+  anchorElements.forEach((anchor) => {
+    anchor.classList.add('social-share');
+    if (anchor.href) {
+      anchor.dataset.href = anchor.href;
+      anchor.removeAttribute('href');
+    }
+    anchor.addEventListener('click', () => handleSocialShareClick(anchor));
+  });
+}
+
+/**
+ * Adds the 'discover-more' class to the specified anchor elements
+ * if their text content contains "Discover More".
+ * @param {Array<Element>} anchorElements - The anchor elements to decorate.
+ * @returns {void}
+ */
+function decorateDiscoverMore(anchorElements) {
+  anchorElements.forEach((anchor) => {
+    anchor.classList.add('discover-more');
+  });
+}
+
+/**
+ * Decorates anchor elements for styling updates via CSS.
+ * @param {Element} [element=document] - The element to decorate.
+ * @returns {void}
+ */
+function decorateAnchors(element = document) {
```

Comment:
This can be moved to block-utils https://github.com/aemsites/icicidirect/blob/main/scripts/blocks-utils.js ?

Reply:
I think it makes sense to keep page rendering lifecycle functions such as the global decorators etc.  in scripts.js